src/provider/client/index.ts

```
@@ -205,37 +206,60 @@ export class QualysAPIClient {
   }
 
   public async verifyAuthentication(): Promise<void> {
-    const endpoint = '/api/2.0/fo/activity_log/';
+    const { getPath, method } = ENDPOINTS.verifyAuthentication;
+    const path = getPath();
 
     try {
       await this.executeAuthenticatedAPIRequest(
-        this.qualysUrl(endpoint, {
+        this.qualysUrl(path, {
           action: 'list',
           username: this.config.qualysUsername,
           truncation_limit: 1,
         }),
         {
-          method: 'GET',
+          method,
         },
       );
     } catch (err) {
       throw new IntegrationProviderAuthenticationError({
         cause: err,
-        endpoint,
+        endpoint: path,
         status: err.status,
         statusText: err.statusText,
       });
     }
   }
 
+  public async verifyApiExistance(): Promise<void> {
+    for (const name in ENDPOINTS) {
+      try {
+        const { getPath, method } = ENDPOINTS[name];
+        const path = getPath(); // uses 'undefined' for path params
+        await this.executeAuthenticatedAPIRequest(this.qualysUrl(path), {
+          method,
+        });
+      } catch (err) {
+        if (err.status == 404) {
```

Don't know if we want to do anything more special than this or not. I also don't know if we should print out a more explicit error message.

---

Looks like we're just letting this silently fail if `err.status != 404`. It seems like we should actually throw an error irrespective of the error status.

Additionally, since this is a verification endpoint, we might want to throw something that returns `true` for `isUserConfigError` or `isProviderAuthenticationError`, because those will be exposed in the Integration Job Log.

---

Updated